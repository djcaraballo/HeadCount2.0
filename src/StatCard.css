.stat-card {
  -moz-box-shadow: 0 10px 6px -6px #777;
  -webkit-box-shadow: 0 10px 6px -6px #777;
  border: 1px solid #ebebeb;
  border-radius: 1rem;
  box-shadow: 0 10px 6px -6px #777;
  box-sizing: border-box;
  height: 20rem;
  width: 15rem;
}

li {
  list-style-type: none;
  line-height: 1.5em;
  font-size: 1em;
}

.greaterThan {
  color: #038198;
  /*font-weight: bold;*/
}

.lessThan {
  color: #ec03ac;
  font-style: italic;
}

.district-name {
  color: #525252;
}