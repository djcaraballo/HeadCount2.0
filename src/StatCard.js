import React from 'react';
import PropTypes from 'prop-types';

const StatCard = (districtData) => {
  return(
    <div>
      <h2>District</h2>
      
    </div>
  )
}

StatCard.propTypes = {
  districtData: PropTypes.array.isRequired
}

export default StatCard;