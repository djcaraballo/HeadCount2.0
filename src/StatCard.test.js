import React from 'react';
import { shallow, expect, test, describe } from 'enzyme';

import StatCard from './StatCard';

describe('StatCard component', () => {
  test('it should match the snapshot', () => {
    const mockDistrictName = 'Denver';
    const mockDistrictStats = [{ 2004: 0.123, 2005: 0.456, 2006: 0.789 }];
    const expectedId = 12345;
    const wrapper = shallow(
      <StatCard
        districtName={mockDistrictName}
        districtStats={mockDistrictStats}
        id={expectedId}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
