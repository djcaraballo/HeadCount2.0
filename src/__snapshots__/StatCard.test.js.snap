// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard component it should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatCard
    districtName="Denver"
    districtStats={
      Array [
        Object {
          "2004": 0.123,
          "2005": 0.456,
          "2006": 0.789,
        },
      ]
    }
    id={12345}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "The StatCard is in the building!",
    },
    "ref": null,
    "rendered": "The StatCard is in the building!",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "The StatCard is in the building!",
      },
      "ref": null,
      "rendered": "The StatCard is in the building!",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
